src/python/pants/help/help_info_extracter.py

```diff
@@ -529,6 +536,14 @@ def load() -> TargetTypeHelpInfo:
             }
         )
 
+        env_var_to_help_info = LazyFrozenDict(
```

Comment:
IMO this should actually be a `FrozenDict`, but one can't be instantiated from `OptionHelpInfo` because it contains mutable values (`lists`). this feels like a deeper issue that should be dealt with elsewhere (addendum: it doesn't seem okay that `LazyFrozenDict`'s lazy values can be mutable?)